packages/core/lib/get-permissions.ts

```diff
@@ -0,0 +1,38 @@
+import { Config, DefaultComponentProps, Permissions } from "../types/Config";
+
+export const getPermissions = ({
+  componentType,
+  globalPermissions,
+  config,
+}: {
+  componentType: keyof DefaultComponentProps;
+  globalPermissions: Partial<Permissions>;
+  config: Config;
+}) => {
+  let componentPermissions = getInitialPermissions({
+    componentType,
+    config,
+    globalPermissions,
+  });
+
+  return componentPermissions;
+};
+
+export const getInitialPermissions = ({
+  componentType,
+  globalPermissions,
+  config,
+}: {
+  componentType: keyof DefaultComponentProps;
+  globalPermissions: Partial<Permissions>;
+  config: Config;
+}) => {
+  let componentPermissions: Partial<Permissions> = { ...globalPermissions };
+
+  componentPermissions = {
+    ...componentPermissions,
+    ...config.components[componentType].permissions,
+  };
+
+  return componentPermissions;
```

Comment:
Feel like there's no need to use a `let` here and you could just return the object directly

```suggestion
   return {
     ...componentPermissions,
     ...config.components[componentType].permissions,
   };
```